plugins/services/src/js/data/groups/resolvers.ts

```diff
@@ -25,12 +26,28 @@ const isGroupArgs = (
 function processServiceGroup(serviceTree: ServiceTree): ServiceGroup {
   return {
     id: serviceTree.getId(),
+    name: serviceTree.getName(),
     quota: {
       enforced: serviceTree.getEnforceRole() === true
     }
   };
 }
 
+function processServiceGroupFilter(
```

Comment:
i personally don't like the mentality of that function. it claims to work for unknown filter-args, so the signature does not tell you what it is that the function wants. when looking at the implementation it looks like it definitely wants a `string`. when looking further it looks like that string must be `JSON.parse`able or else you'll just get your groups back.

this of course is not a big thing on its own but i think we should try to not introduce code that's dealing with so much uncertainty regarding input and output as stuff tends to get hard to trace down in case a debugging is necessary.

🙈

Reply:
We could have a larger conversation around structuring graphql schema and arguments for the data layer. The basic trade-off I was trying to meet here was generic vs specific schema. If we went with a more specific schema for quota, then having more specific arguments would make sense. I opt-ed for a more general schema that can be grown, which then IMO lends itself to more generic arguments. As these seems to be what makes graphQL useful IMO. To me having a bunch of specific queries doesn't add much value over just having stores with functions.

Reply:
my comment is not about graphQL at all but only about the second argument that can be unknown but is only handled if it's a `JSON.parse`-able string.

if that's related and i don't see that because i'm missing some part of the bigger picture here, please ignore my comment.

Reply:
That argument is coming from directly from GraphQL arguments which are `unknown`, we could do the type check in the resolver but I thought it better to just pass it down. 🤷‍♂

Reply:
ah! thanks for pointing that out! 🤦‍♂

i'll have another look then and am perfectly fine with this till then.